org.libretro.RetroArch.json.tmpl

```diff
@@ -0,0 +1,191 @@
+    "--share=ipc",
+    "--share=network",
+    "--device=all",
+    "--filesystem=host",
+    "--filesystem=home",
+    "--allow=multiarch",
+    "--talk-name=org.freedesktop.ScreenSaver",
+    "--talk-name=org.freedesktop.PowerManagement.Inhibit",
+    "--talk-name=org.freedesktop.login1",
+    "--filesystem=xdg-run/app/com.discordapp.Discord:create"
+  ],
+  "modules": [
+    {
+      "name": "retroarch",
+      "config-opts": [
+        "--enable-dbus"
+      ],
+      "make-args": [
+        "GLOBAL_CONFIG_DIR=${FLATPAK_DEST}/etc"
+      ],
+      "sources": [
+        {
+          "type": "git",
+          "url": "https://github.com/libretro/RetroArch.git",
+          "tag": "%%VERSION_TAG%%"
+        },
+        {
+          "type": "file",
+          "path": "org.libretro.RetroArch.appdata.xml"
+        },
+        {
+          "type": "file",
+          "path": "retroarch.cfg"
+        }
+      ],
+      "post-install": [
+        "mkdir -p ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/",
+        "mv ${FLATPAK_DEST}/share/pixmaps/retroarch.svg ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/",
+        "rmdir ${FLATPAK_DEST}/share/pixmaps/",
+        "mkdir -p ${FLATPAK_DEST}/etc",
+        "sed s:@prefix@:${FLATPAK_DEST}:g retroarch.cfg > ${FLATPAK_DEST}/etc/retroarch.cfg",
+        "mkdir -p ${FLATPAK_DEST}/share/appdata",
+        "cp org.libretro.RetroArch.appdata.xml ${FLATPAK_DEST}/share/appdata"
+      ],
+      "modules": [
+        "libpng/libpng-1.6.35.json",
+        "nvidia-cg-toolkit/nvidia-cg-toolkit-3.1.0013.json",
+        "shared-modules/SDL/SDL-1.2.15.json",
+        "shared-modules/SDL/SDL_image-1.2.12.json",
+        "shared-modules/SDL/SDL_mixer-1.2.12.json",
+        "shared-modules/SDL/SDL_net-1.2.8.json",
+        "shared-modules/SDL/SDL_ttf-2.0.11.json",
+        "shared-modules/libusb/libusb.json",
+        "shared-modules/gudev/gudev.json",
+        "libbz2/libbz2-1.0.8.json",
+        "xrandr/xrandr-1.5.1.json"
+      ]
+    },
+    {
+      "name": "retroarch-filers-video",
+      "subdir": "gfx/video_filters",
+      "make-install-args": [
+        "PREFIX=${FLATPAK_DEST}"
+      ],
+      "sources": [
+        {
+          "type": "git",
+          "url": "https://github.com/libretro/RetroArch.git",
+          "tag": "%%VERSION_TAG%%"
+        }
+      ]
+    },
+    {
+      "name": "retroarch-filers-audio",
+      "subdir": "libretro-common/audio/dsp_filters",
+      "make-install-args": [
+        "PREFIX=${FLATPAK_DEST}"
+      ],
+      "sources": [
+        {
+          "type": "git",
+          "url": "https://github.com/libretro/RetroArch.git",
+          "tag": "%%VERSION_TAG%%"
+        }
+      ]
+    },
+    {
+      "name": "retroarch-assets",
+      "make-install-args": [
+        "PREFIX=${FLATPAK_DEST}"
+      ],
+      "sources": [
+        {
+          "type": "git",
+          "url": "https://github.com/libretro/retroarch-assets.git",
+          "branch": "master"
```

Comment:
This is a fantastic change! Makes updating it so much easier. Only minor concern I have is targeting master for some of the supporting repositories. There is no guarantee that `master` will always be good for the given version. 🤔

Reply:
Hi @RobLoach , I've posted a comment on issue #189 about what I'm trying to do, I guess you might have turned off notifications for that issue, I can @you in the future if you prefer that. This thing is obviously not ready yet, very much WIP state, I wouldn't leave those refs to master there.